pkg/lib/catalogsource/catalogsource_update.go

```diff
@@ -84,17 +84,24 @@ even if the provided conditions result in no changes.
 • conditions: condition values to be updated
 */
 func UpdateSpecAndStatusConditions(logger *logrus.Entry, client versioned.Interface, catsrc *v1alpha1.CatalogSource, conditions ...metav1.Condition) error {
-
-	// update the conditions
-	for _, condition := range conditions {
-		meta.SetStatusCondition(&catsrc.Status.Conditions, condition)
-	}
-
 	// make the update if possible
-	if _, err := client.OperatorsV1alpha1().CatalogSources(catsrc.GetNamespace()).Update(context.TODO(), catsrc, metav1.UpdateOptions{}); err != nil {
+	catsrc, err := client.OperatorsV1alpha1().CatalogSources(catsrc.GetNamespace()).Update(context.TODO(), catsrc, metav1.UpdateOptions{})
+	if err != nil {
 		logger.WithError(err).Error("UpdateSpecAndStatusConditions - unable to update CatalogSource image reference")
 		return err
 	}
+
+	if conditions != nil && len(conditions) > 0 {
+		// update the conditions
+		for _, condition := range conditions {
+			meta.SetStatusCondition(&catsrc.Status.Conditions, condition)
+		}
+
+		if _, err := client.OperatorsV1alpha1().CatalogSources(catsrc.GetNamespace()).UpdateStatus(context.TODO(), catsrc, metav1.UpdateOptions{}); err != nil {
+			logger.WithError(err).Error("UpdateSpecAndStatusConditions - unable to update CatalogSource status condition")
+			return err
+		}
```

Comment:
I have just a real quick question: any idea on whether firing off the status sub-resource update first, before updating the catalogsource resource would bump the resourceVersion in the client-go world? Maybe I'm muddling this behavior in controller-runtime, where an update to the status sub-resource doesn't cause a new reconciliation.

Reply:
The `resourceVersion` should change even if only the status is being updated. I expect the reconciliation to be triggered for any changes that happen to the catalogsources though that's expected I would say. This code is a bug fix.